StitchCoreServicesMongoDbRemote/Sources/StitchCoreServicesMongoDbRemote/Internal/CoreRemoteMongoCollection.swift

```diff
@@ -0,0 +1,283 @@
+import Foundation
+import MongoSwift
+import StitchCore
+
+public class CoreRemoteMongoCollection<T: Codable> {
+    /**
+     * A `Codable` type associated with this `MongoCollection` instance.
+     * This allows `CollectionType` values to be directly inserted into and
+     * retrieved from the collection, by encoding/decoding them using the
+     * `BsonEncoder` and `BsonDecoder`.
+     * This type association only exists in the context of this particular
+     * `MongoCollection` instance. It is the responsibility of the user to
+     * ensure that any data already stored in the collection was encoded
+     * from this same type.
+     */
+    public typealias CollectionType = T
+    
+    /**
+     * The name of this collection.
+     */
+    public let name: String
+    
+    /**
+     * The name of the database containing this collection.
+     */
+    public let databaseName: String
+    
+    private var baseOperationArgs: Document {
```

Comment:
[minor] This should be `lazy` or we should be using this differently. We're operating on this as if it's a set value, but it's being calculated each time.